pkg/cmd/server/options.go

```diff
@@ -14,51 +14,75 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package server provides configuration options for the metrics API server.
 package server
 
 import (
 	"fmt"
 	"net"
 
+	"github.com/spf13/pflag"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	genericoptions "k8s.io/apiserver/pkg/server/options"
 	openapicommon "k8s.io/kube-openapi/pkg/common"
 
 	"sigs.k8s.io/custom-metrics-apiserver/pkg/apiserver"
 )
 
+// CustomMetricsAdapterServerOptions contains the of options used to configure
+// the metrics API server.
+//
+// It is based on a subset of [genericoptions.RecommendedOptions].
 type CustomMetricsAdapterServerOptions struct {
-	// genericoptions.ReccomendedOptions - EtcdOptions
 	SecureServing  *genericoptions.SecureServingOptionsWithLoopback
 	Authentication *genericoptions.DelegatingAuthenticationOptions
 	Authorization  *genericoptions.DelegatingAuthorizationOptions
 	Audit          *genericoptions.AuditOptions
 	Features       *genericoptions.FeatureOptions
 
-	// OpenAPIConfig
 	OpenAPIConfig *openapicommon.Config
+	EnableMetrics bool
 }
 
+// NewCustomMetricsAdapterServerOptions creates a new instance of
+// CustomMetricsAdapterServerOptions with its default values.
 func NewCustomMetricsAdapterServerOptions() *CustomMetricsAdapterServerOptions {
 	o := &CustomMetricsAdapterServerOptions{
 		SecureServing:  genericoptions.NewSecureServingOptions().WithLoopback(),
 		Authentication: genericoptions.NewDelegatingAuthenticationOptions(),
 		Authorization:  genericoptions.NewDelegatingAuthorizationOptions(),
 		Audit:          genericoptions.NewAuditOptions(),
 		Features:       genericoptions.NewFeatureOptions(),
+
+		EnableMetrics: true,
 	}
 
 	return o
 }
 
-func (o CustomMetricsAdapterServerOptions) Validate(args []string) error {
-	return nil
+// Validate validates CustomMetricsAdapterServerOptions
+func (o CustomMetricsAdapterServerOptions) Validate() error {
+	errors := []error{}
+	errors = append(errors, o.SecureServing.Validate()...)
+	errors = append(errors, o.Authentication.Validate()...)
+	errors = append(errors, o.Authorization.Validate()...)
+	errors = append(errors, o.Audit.Validate()...)
+	errors = append(errors, o.Features.Validate()...)
+	return utilerrors.NewAggregate(errors)
 }
 
-func (o *CustomMetricsAdapterServerOptions) Complete() error {
-	return nil
+// AddFlags adds the flags defined for the options, to the given flagset.
+func (o *CustomMetricsAdapterServerOptions) AddFlags(fs *pflag.FlagSet) {
+	o.SecureServing.AddFlags(fs)
+	o.Authentication.AddFlags(fs)
+	o.Authorization.AddFlags(fs)
+	o.Audit.AddFlags(fs)
+	o.Features.AddFlags(fs)
 }
 
+// Config returns configuration for the API server given CustomMetricsAdapterServerOptions
 func (o CustomMetricsAdapterServerOptions) Config() (*apiserver.Config, error) {
```

Comment:
Since this is meant to be consumed as a library, I would be in favor of updating this method to follow a similar workflow as the options from the kube-apiserver recommended options. The signature would look like:
```go
func (o CustomMetricsAdapterServerOptions) ApplyTo(config *apiserver.Config) error
```